pig/src/main/java/com/netflix/iceberg/pig/IcebergInputFormat.java

```diff
@@ -0,0 +1,260 @@
+import com.netflix.iceberg.hadoop.HadoopInputFile;
+import com.netflix.iceberg.io.InputFile;
+import com.netflix.iceberg.parquet.Parquet;
+import com.netflix.iceberg.types.TypeUtil;
+import com.netflix.iceberg.types.Types;
+import org.apache.commons.lang.SerializationUtils;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.parquet.hadoop.api.ReadSupport;
+import org.apache.pig.impl.util.ObjectSerializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static com.netflix.iceberg.pig.SchemaUtil.*;
+
+public class IcebergInputFormat<T> extends InputFormat<Void, T> {
+  private static final Logger LOG = LoggerFactory.getLogger(IcebergInputFormat.class);
+
+  public static final String ICEBERG_SCHEMA = "iceberg.schema";
+  public static final String ICEBERG_PROJECTED_FIELDS = "iceberg.projected.fields";
+  public static final String ICEBERG_FILTER_EXPRESSION = "iceberg.filter.expression";
+
+  private static final List<String> SNAPSHOT_COLUMNS = Arrays.asList(
+      "snapshot_id", "file_path", "file_ordinal", "file_format", "block_size_in_bytes",
+      "file_size_in_bytes", "record_count", "partition", "value_counts", "null_value_counts",
+      "lower_bounds", "upper_bounds"
+  );
+
+  private Table table;
+  private List<InputSplit> splits;
+
+  public IcebergInputFormat(Table table) {
+    this.table = table;
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public List<InputSplit> getSplits(JobContext context) throws IOException, InterruptedException {
+    if (splits != null) {
+      LOG.info("Returning cached splits: " + splits.size());
+      return splits;
+    }
+
+    splits = Lists.newArrayList();
+
+    TableScan scan = table.newScan().select(SNAPSHOT_COLUMNS);
+
+    //Apply Filters
+    Expression filterExpression = (Expression) ObjectSerializer.deserialize(context.getConfiguration().get(ICEBERG_FILTER_EXPRESSION));
+
+    if (filterExpression != null) {
+      LOG.info("Filter Expression: " + filterExpression);
+      scan = scan.filter(filterExpression);
+    }
+
+    //Wrap in Splits
+    scan.planTasks().forEach((scanTask) -> splits.add(new IcebergSplit(scanTask)));
+
+    return splits;
+  }
+
+  @Override
+  public RecordReader<Void, T> createRecordReader(InputSplit split, TaskAttemptContext context) throws IOException, InterruptedException {
+    return new IcebergRecordReader<>();
+  }
+
+  public static class IcebergSplit extends InputSplit implements Writable {
+    private CombinedScanTask task;
```

Comment:
This is a public class, but without a getter for the task it is useless for other purposes. This uses the field, but that can't be done outside of this file.

This should either be a private class or it should expose a getter for the task. I'd prefer the first approach since we don't fully support Iceberg through an InputFormat yet.

Reply:
Agree, will make private.